dev/tools/localization/localizations_utils.dart

```diff
@@ -377,32 +377,29 @@ class $classNamePrefix$camelCaseName extends $superClass {''';
 /// foo\nbar => 'foo\\\\nbar'
 /// ```
 ///
-/// When [shouldEscapeDollar] is set to true, the
-/// result avoids character escaping, with the
-/// exception of the dollar sign:
+/// This function is used by tools that take in a JSON-formatted file to
+/// generate Dart code. For this reason, characters with special meaning
+/// in JSON files, such as the backspace character (\b) have to be
+/// properly escaped by this function so that the generated Dart code
+/// correctly represents this character.
+///
+/// When [shouldEscapeDollar] is set to true, the result avoids character
```

Comment:
I'm not what this means; doesn't sound right. It should just mean that '$' characters in [value] will be replaced by '\$' in the returned string.